src/createReducer.test.ts

```diff
@@ -106,6 +106,105 @@ describe('createReducer', () => {
     behavesLikeReducer(wrappedReducer)
   })
 
+  describe('actionMatchers argument', () => {
+    const prepareNumberAction = (payload: number) => ({
+      payload,
+      meta: { type: 'number_action' }
+    })
+    const prepareStringAction = (payload: string) => ({
+      payload,
+      meta: { type: 'string_action' }
+    })
+
+    const numberActionMatcher = (a: AnyAction): a is PayloadAction<number> =>
+      a.meta.type === 'number_action'
+    const stringActionMatcher = (a: AnyAction): a is PayloadAction<string> =>
+      a.meta.type === 'string_action'
+
+    const incrementBy = createAction('increment', prepareNumberAction)
+    const decrementBy = createAction('decrement', prepareNumberAction)
+    const concatWith = createAction('concat', prepareStringAction)
+
+    const initialState = { numberActions: 0, stringActions: 0 }
+    const numberActionsCounter = {
+      matcher: numberActionMatcher,
+      reducer(state: typeof initialState) {
+        state.numberActions = state.numberActions * 10 + 1
+      }
+    }
+    const stringActionsCounter = {
+      matcher: stringActionMatcher,
+      reducer(state: typeof initialState) {
```

Comment:
I don't see any test cases that use `action` here. What's the inferred action type like, if any?

Reply:
Over in the typetest in https://github.com/reduxjs/redux-toolkit/pull/610/files#diff-8e07add1ff60e559fa564decce5eb329R66

Here, it would be `PayloadAction<string>` due to the declaration of `stringActionMatcher`.